### PR TITLE
Send relevant email on FM Activity actions

### PR DESCRIPTION
**Pull request by reinbach**
[ch21567]
[ch21571]
[ch21573]
[ch21569]
[ch21572]
[ch21568]

### CODE REVIEW

**Comment by robertavram** [comment]
Can you not use side effects?

**Comment by robertavram** [comment]
Can you not use side effects?

Just read your comment. I thought direct transitions also send side effects. As long as the validator is called.

**Comment by reinbach** [comment]
None of the transition methods appear to be called. Still looking into why not, and what I'm missing.

File: src/etools/applications/field_monitoring/planning/models.py [hunk 1]
```diff
@@ -343,9 +387,20 @@ def auto_accept_staff_activity(self):
         if self.monitor_type == self.MONITOR_TYPE_CHOICES.staff:
             self.accept()
             self.save()
-            # todo: direct transitions doesn't trigger side effects. trigger effects manually? or rewrite this effect?
+            # todo: direct transitions doesn't trigger side effects.
+            # trigger effects manually? or rewrite this effect?
             self.init_offline_blueprints()
 
+        # send email to users assigned to fm activity
+        recipients = list(self.team_members.all()) + [self.person_responsible]
```
Comment (roman-karpovich) [inline src/etools/applications/field_monitoring/planning/models.py:395]: maybe use set here to avoid possible email duplication

Comment (reinbach) [reply src/etools/applications/field_monitoring/planning/models.py]: :+1:

**Comment by robertavram** [comment]
The way we implemented it, it's not designed to call the functions decorated functions. So nothing in the functions gets executed unless you call the function itself, if you do that, you'll transition but effectively circumvent the validator. I would not place any content in the transition functions and use side effects. Although, not completely sure how it was implemented in fm

**Comment by reinbach** [comment]
I keep forgetting that the validator is handling it :(

**Comment by roman-karpovich** [comment]
> None of the transition methods appear to be called. Still looking into why not, and what I'm missing.

they are defined here just for test purposes. in fact side effects are to be used for treal ransitions

**Pull request merged**